In [2]:
import torch
import torch.nn as nn
import torchvision
import torchvision.transforms as transforms
import numpy as np
import matplotlib.pyplot as plt
import time


In [3]:
device = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')

batch_size = 70

data_transforms = transforms.Compose([
    transforms.Resize(512),
    transforms.CenterCrop(512),
    transforms.ToTensor(),
    transforms.Normalize(mean=[0.485, 0.456, 0.406],
                         std=[0.229, 0.224, 0.225])
])

train_data = torchvision.datasets.ImageFolder(root='./animals/train',
                                                 transform=data_transforms)

train_loader = torch.utils.data.DataLoader(train_data, batch_size=batch_size,
                                           shuffle=True, num_workers=8)


print(f"Количество классов {train_data.classes} Количестов примеров  {len(train_data.samples)}")
print(device)

class_names = train_data.classes

Количество классов ['cat', 'dog', 'wild'] Количестов примеров  14630
cuda:0


In [4]:
test_dataset = torchvision.datasets.ImageFolder(root='./animals/val',
                                             transform=data_transforms)
test_loader  = torch.utils.data.DataLoader(test_dataset, batch_size=batch_size,
                                    shuffle=True, num_workers=8)

print(f"Количество классов {test_dataset.classes} Количестов примеров  {len(test_dataset.samples)}")


Количество классов ['cat', 'dog', 'wild'] Количестов примеров  1500


torch.Size([70, 3, 512, 512])

E:\Maga\1.2\env\Scripts\python.exe E:\Maga\2_1\deepLearning\proj\lab4\cnnDogsAndCats.py 
Количество классов ['cat', 'dog', 'wild'] Количестов примеров  14630
cuda:0
Количество классов ['cat', 'dog', 'wild'] Количестов примеров  1500
torch.Size([70, 3, 512, 512])
Эпоха 0 из 3 Шаг 0 Ошибка:  1.0989346504211426
Эпоха 0 из 3 Шаг 100 Ошибка:  0.5985499620437622
Эпоха 0 из 3 Шаг 200 Ошибка:  0.49503418803215027
Эпоха 1 из 3 Шаг 0 Ошибка:  0.3429974913597107
Эпоха 1 из 3 Шаг 100 Ошибка:  0.2544006407260895
Эпоха 1 из 3 Шаг 200 Ошибка:  0.34183958172798157
Эпоха 2 из 3 Шаг 0 Ошибка:  0.27612414956092834
Эпоха 2 из 3 Шаг 100 Ошибка:  0.19264349341392517
Эпоха 2 из 3 Шаг 200 Ошибка:  0.1279948502779007
2241.9502782821655
Точность модели: 91.26666666666667%

Process finished with exit code 0


C:\Users\Suvorov\AppData\Local\Temp\ipykernel_158916\2246367131.py:4: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  net.load_state_dict(torch.load('animals.ckpt'))


RuntimeError: Input type (torch.cuda.FloatTensor) and weight type (torch.FloatTensor) should be the same